### PR TITLE
[PUBDEV-3278] [PUBDEV-3277] [PUBDEV-3430] [PUBDEV-3440] Changes to H2OFrame setter

### PR DESCRIPTION
**Pull request by st-pasha**
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/220)

### CODE REVIEW

File: h2o-py/h2o/frame.py [hunk 1]
```diff
@@ -1093,11 +1093,6 @@ def __getitem__(self, item):
         fr._ex._cache.types = new_types
         return fr
 
-    # def __getattr__(self, key):
```
Comment (navdeep-G) [inline h2o-py/h2o/frame.py]: Why is this deleted?

File: h2o-py/h2o/frame.py [hunk 2]
```diff
@@ -685,7 +685,8 @@ def mktime(year=1970, month=0, day=0, hour=0, minute=0, second=0, msec=0):
         -------
           H2OFrame of one column containing the date in millis since the epoch.
         """
-        return ExprNode("mktime", year, month, day, hour, minute, second, msec)._eager_frame()
+        return H2OFrame._expr(ExprNode("mktime", year, month, day, hour, minute, second, msec))
```
Comment (navdeep-G) [inline h2o-py/h2o/frame.py]: Good catch

**Comment by navdeep-G** [comment]
Looks good.

File: h2o-py/h2o/frame.py [hunk 3]
```diff
@@ -171,6 +173,10 @@ def get_frame(frame_id):
             return None
         return fr
 
+    def refresh(self):
```
Comment (navdeep-G) [inline h2o-py/h2o/frame.py]: Is the idea a user would do self.refresh() to get their frame back to what it was imported as?

Comment (st-pasha) [inline h2o-py/h2o/frame.py]: It's more of a debug/test function: whenever you call any function on a frame, it may not be immediately executed, but instead stored as a `frame._ex`. At the same time, frame's properties (such as list of columns, their types, frame's shape, etc) will be pre-computed by the client and stored with the frame. This may potentially cause a discrepancy between the client and the server: the client may have computed those properties incorrectly. This function then can be used to fix these issues and reload the frame's properties from the server. It may also be used to eagerize the frame if needed.

This is probably not the best choice of a name for a function, and I'm not yet sure whether this should be public or private...

File: h2o-py/h2o/frame.py (hunk 1 above)
Comment (st-pasha) [inline h2o-py/h2o/frame.py]: Well, at one point I thought to implement the same functionality they have in pandas: so that for example `frame.C1` can be used to refer to a column "C1" within the frame. I no longer think it's a good idea (and pandas guys probably would agree with me). The problem with this approach is that: (1) this syntax can only be used to access columns that are not any of the existing properties / methods on the H2OFrame (thus the user is expected to know what those 200+ names are), (2) any subsequent change to the frame's API, including adding new methods, will become a breaking change because you are making new names into reserved ones...

Comment (st-pasha) [inline h2o-py/h2o/frame.py]: There could be a middle-ground solution for example if we allow such access for column names starting with a capital letter, and make a rule never to add any methods that start with a capital letter -- but I'm not sure how useful this will be.

Comment (mmalohlava) [inline h2o-py/h2o/frame.py]: Just naive idea following ^^: we can say that rule is `frame.COL_<actual colname>`

But what if colname would contain string like "average+age+ca" (this is from census dataset)?
